pkg/util/limiter/query_limiter.go

```diff
@@ -16,6 +16,16 @@ import (
 	"github.com/grafana/mimir/pkg/util/validation"
 )
 
+type MemoryTracker interface {
+	IncreaseMemoryConsumptionForLabels(labels labels.Labels) error
+}
+
+type NoopMemoryTracker struct{}
```

Comment:
This isn't needed, we can create a `MemoryConsumptionTracker` with limit set to 0 to disable enforcing a memory consumption limit.